### PR TITLE
chudley-elliot fit function should contain hbar

### LINKED ISSUE #22694: chudley-elliot fit function should contain hbar
A user complained that the lack of hbar in the [Chudley-Elliot](http://docs.mantidproject.org/nightly/fitting/fitfunctions/ChudleyElliot.html) model had confused him and his fits for a long time. It should be incorporated to the model

### PR DESCRIPTION
**Pull request by jmborr**
- [x] Updated source
- [x] Unit test
- [x] Updated docs
- [x] Release notes

**To test:**
Run the following code in Mantid:
```
from __future__ import print_function
import numpy as np
from scipy import constants
import math

# Create Input workspace, a ChudleyElliot HWHM function with Tau=1.42, and L=2.42
planck_constant = constants.Planck / constants.e * 1E15  # meV*psec
hbar = planck_constant / (2 * np.pi)  # meV * ps  = ueV * ns
q = np.asarray([0.01, 0.1, 1.0, 10.0])
hwhm = hbar * (1.0 - np.sin(2.42 * q) / (2.42 * q)) / 1.42
w = CreateWorkspace(q, hwhm, hwhm * 0.1)

# Fit and then check optimal parameters against target parameters. A few
# percent difference is fine
fit = Fit(Function=ChudleyElliot(), InputWorkspace=w, Minimizer='Simplex')
print([fit.Function[p] for p in ('Tau', 'L')])  # compare to 1.42 and 2.42
```

Fixes #22694

Does this update require release notes?
- [x] Yes [Updated Release Notes](https://github.com/mantidproject/mantid/pull/22732/files#diff-d41279afe39cfbf234c3ca92983d1c5cR96)
- [ ] No

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by keeeto — CHANGES_REQUESTED** [review]
This looks mostly good to me - the prescribed test gives the results as described. There are just a few minor points I would raise.

File: docs/source/fitting/fitfunctions/ChudleyElliot.rst [hunk 1]
```diff
@@ -15,7 +15,13 @@ associated long range diffusive motions of molecules.
 
 The Chudley-Elliot Jump diffusion model [1]_ has the form:
 
-.. math:: Gamma(Q) = (1 - sin(Ql)/Ql)/tau
+.. math:: HWHM(Q) = \frac{\hbar}{\tau} \cdot \frac{1 - sin(Ql)}{Ql}
+
+Units of :math:`l` are inverse units of :math:`Q`.
+
+Units of :math:`` are :math:`meV` if units of :math:`\tau` are *ps*.
```
Comment (keeeto) [inline docs/source/fitting/fitfunctions/ChudleyElliot.rst]: Should this be Units of :math:`HWHM`?

File: Framework/PythonInterface/plugins/functions/ChudleyElliot.py [hunk 2]
```diff
@@ -44,7 +48,7 @@ def function1D(self, xvals):
         length = self.getParameterValue("L")
 
         xvals = np.array(xvals)
-        hwhm = (1.0 - np.sin(xvals * length) / (xvals * length)) / tau
+        hwhm = self.hbar*(1.0 - np.sin(xvals * length) / (xvals * length))/tau
```
Comment (keeeto) [inline Framework/PythonInterface/plugins/functions/ChudleyElliot.py]: There is some mixing of whitespace conventions in this equation. Should the operators / and * have either surrounding whitespace or no surrounding whitespace, but not both? There could be good reason for this and I am mistaken!

File: Framework/PythonInterface/plugins/functions/ChudleyElliot.py [hunk 3]
```diff
@@ -54,9 +58,9 @@ def functionDeriv1D(self, xvals, jacobian):
         i = 0
         for x in xvals:
             s = math.sin(x*length)/(x*length)
-            h = (1.0-s)/tau
+            h = self.hbar * (1.0-s)/tau
             jacobian.set(i,0,-h/tau)
-            jacobian.set(i,1,(math.cos(x*length)-s)/(length*tau))
+            jacobian.set(i,1, self.hbar * (math.cos(x*length)-s)/(length*tau))
```
Comment (keeeto) [inline Framework/PythonInterface/plugins/functions/ChudleyElliot.py]: A similar point to the previous one about whitespace and operators.

**Review by keeeto — APPROVED** [review]
All comments are minor and the PR otherwise looks good. The functional test passes.

File: Framework/PythonInterface/plugins/functions/ChudleyElliot.py [hunk 4]
```diff
@@ -27,10 +27,14 @@
 import math
 import numpy as np
 from mantid.api import IFunction1D, FunctionFactory
+from scipy import constants
 
 
 class ChudleyElliot(IFunction1D):
 
+    planck_constant = constants.Planck / constants.e * 1E15  # meV*psec
```
Comment (DanNixon) [inline Framework/PythonInterface/plugins/functions/ChudleyElliot.py]: `constants.value('Planck constant over 2 pi in eV s') * 1e15` may be a cleaner way to express this.

Comment (keeeto) [reply Framework/PythonInterface/plugins/functions/ChudleyElliot.py]: That is cleaner. I didn't know about the natural language feature of scipy constants, nice.

**Comment by jmborr** [comment]
Fixing the system test is going to have to wait until I install Ubuntu 18.04 since I can't build Mantid in 16.04 anymore, and I am not in the mood to figure out why :unamused:

**Comment by jmborr** [comment]
arrggg I updated this with master and now it's a total screw up

**Comment by jmborr** [comment]
I don't know what happened with the million commits, so I'll close this pull request and reopen a pull request with a more sane branch

**Pull request closed without merging**